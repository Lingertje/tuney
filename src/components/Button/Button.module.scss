@use '../../css/config' as *;

.button {
    padding: 8px 16px;
    line-height: 1;
    font-weight: 500;
    border: none;
    background-color: transparent;
    cursor: pointer;

    &--primary {
        color: #fff;
        border: 2px solid #fff;
        border-radius: 3px;
        transition: all 200ms ease-out;

        &:hover {
            color: $color-primary;
            background-color: #fff;
        }
    }

    &--secondary {

    }

    &--ghost {

    }
}
