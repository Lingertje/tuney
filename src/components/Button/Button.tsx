import { FC, memo } from "react";

import style from "./Button.module.scss";

export enum ButtonTypes {
	PRIMARY = "primary",
	SECONDARY = "secondary",
	GHOST = "ghost"
}

type ButtonProps = {
	type?: ButtonTypes
	handler?: () => unknown,
	disabled?: boolean
	children: React.ReactNode
}


const Button: FC<ButtonProps> = ({ children, type = ButtonTypes.PRIMARY, handler, disabled }: ButtonProps) => {
	const buttonClass = `button--${type}`;

	return (
		<button className={`${style["button"]} ${style[buttonClass]}`} onClick={handler ? handler : undefined} disabled={disabled} >
			{children}
		</button>
	);
};

export default memo(Button);
