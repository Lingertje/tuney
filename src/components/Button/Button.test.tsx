import { render } from "@testing-library/react";

import Button, { ButtonTypes } from "components/Button/Button";

test("Should render a button", () => {
	const { asFragment, getByRole } = render(<Button type={ButtonTypes.PRIMARY}>Click me</Button>);
	expect(getByRole("button")).toBeInTheDocument;
	expect(asFragment()).toMatchSnapshot();
});

test("Should render a disabled button", () => {
	const { asFragment, getByRole } = render(<Button type={ButtonTypes.PRIMARY} disabled>Click me</Button>);
	expect(getByRole("button")).toBeDisabled;
	expect(asFragment()).toMatchSnapshot();
});
