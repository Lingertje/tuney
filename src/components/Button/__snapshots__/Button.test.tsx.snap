// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a button 1`] = `
<DocumentFragment>
  <button
    class="button button--primary"
  >
    Click me
  </button>
</DocumentFragment>
`;

exports[`Should render a disabled button 1`] = `
<DocumentFragment>
  <button
    class="button button--primary"
    disabled=""
  >
    Click me
  </button>
</DocumentFragment>
`;
